hologram_dashboard/fetcher/hologram_data_source_availability_fetcher.cc

```diff
@@ -55,16 +55,98 @@ void HologramDataSourceAvailabilityFetcher::FetchFromDatabase() {
 }
 
 void HologramDataSourceAvailabilityFetcher::GetHologramDataAvailability(
-    absl::Time date) {
-    // TODO(alexanderlin): Implement.
-    // for(const std::pair<std::string, std::string>& it : system_to_cell_map_) {
-    //     for(int i = 0; i < hologram_configs_.data_source_config_size(); ++i) {
-    //     UpdateProto(it.first, date, hologram_configs_.data_source_config(i), status);
-    // }
+    absl::Time time) {
+    absl::TimeZone google_time;
+    assert(absl::LoadTimeZone("America/Los_Angeles", &google_time));
+    // Hardcode database location since there's no access to real Hologram
+    // server.
+    std::filesystem::path database_loc = std::filesystem::current_path();
+    database_loc += "/fetcher/testdata/Database/";
+    std::string date = absl::FormatTime("%Y/%m/%d/", time, google_time);
+    
+    for(const std::pair<System, std::string>& it : system_to_cell_) {
+        for(int i = 0; i < hologram_configs_.data_source_config_size(); ++i) {
+            HologramConfig config = hologram_configs_.data_source_config(i);
+            std::filesystem::path update_coordinator_path = database_loc; 
+            update_coordinator_path += it.second + "/" ;
+            update_coordinator_path += "update_coordinator_done/" + date;
+            std::filesystem::path update_lookup_server_path = database_loc;
+            update_lookup_server_path +=  it.second + "/";
+            update_lookup_server_path += "update_lookup_server_done/" + date;
+            // If corpus has not succeeded for the day yet no data is injested.
+            std::string latest_job = UpdateCorpusFinishTime(
```

Comment:
I think you need the list of update_finish_time?

We may do multiple time Kvick Server update in a day. Usually the first update is the most critical one, instead of the lastest one.